src/gfx_device/display_config.rs

```diff
@@ -1,25 +1,31 @@
 //! Graphical display configuration.
 
-config!(
-    /// Graphical display configuration.
-    ///
-    /// These are fed in when calling `video_init()`.
-    pub struct DisplayConfig {
-        /// Name of the application window.
-        pub title: String = "Amethyst game".to_string(),
-        /// Enables or disables fullscreen mode.
-        pub fullscreen: bool = false,
-        /// Current window dimensions, measured in pixels (px).
-        pub dimensions: Option<(u32, u32)> = None,
-        /// Minimum window dimensions, measured in pixels (px).
-        pub min_dimensions: Option<(u32, u32)> = None,
-        /// Maximum window dimensions, measured in pixels (px).
-        pub max_dimensions: Option<(u32, u32)> = None,
-        /// Enables or disables vertical synchronization.
-        pub vsync: bool = true,
-        /// Level of MSAA anti-aliasing.
-        pub multisampling: u16 = 1,
-        /// Sets the visibility of the window.
-        pub visibility: bool = true,
-    }
-);
+/// Graphical display configuration.
+///
+/// These are fed in when calling `video_init()`.
+#[derive(Debug, Default, Deserialize, Serialize)]
+pub struct DisplayConfig {
+    /// Name of the application window.
+    pub title: String,
+    /// Enables or disables fullscreen mode.
+    #[serde(default)]
+    pub fullscreen: bool,
+    /// Current window dimensions, measured in pixels (px).
+    #[serde(default)]
+    pub dimensions: Option<(u32, u32)>,
+    /// Minimum window dimensions, measured in pixels (px).
+    #[serde(default)]
+    pub min_dimensions: Option<(u32, u32)>,
+    /// Maximum window dimensions, measured in pixels (px).
+    #[serde(default)]
+    pub max_dimensions: Option<(u32, u32)>,
+    /// Enables or disables vertical synchronization.
+    #[serde(default)]
+    pub vsync: bool,
+    /// Level of MSAA anti-aliasing.
+    #[serde(default)]
+    pub multisampling: u16,
+    /// Sets the visibility of the window.
+    #[serde(default = "true")]
```

Comment:
Yeah, I know that it doesn't work yet. For whatever reason it accepts a function name. I'll try to make a PR for `#[serde(default_value=...)]`.